File: Semestre 4/IASA_Aux/IASA/2020_2021/tp2/src/tp2/prof/ProcuraProfIter.java
package tp2.prof;

import tp2.modprob.Problema;
import tp2.pee.Solucao;

public class ProcuraProfIter extends ProcuraProf{
	private int incProf = 0;
	
	public ProcuraProfIter(int incProf) {
		super();
		this.incProf = incProf;	
	}
	
	
	public int getIncProf() {
		return this.incProf;
	}
	
	public void setIncProf(int incProf) {
		this.incProf = incProf;
	}
	
	
	public Solucao resolver(Problema problema, int profMax) {
	
		for(int i = 0;i<profMax;i++) {
			Solucao sol = super.resolver(problema, profMax);
			if(sol!=null) return sol;
		}
		
		System.out.println(4);
		return null;		
	}
}

File: Semestre 4/IASA_Aux/IASA/2022_2023/iasa45741/iasa-agente/src/lib/pee/prof/procura_prof_iter.py
from .procura_prof_lim import ProcuraProfLim
from abc import abstractmethod


class ProcuraProfIter(ProcuraProfLim):
    def resolver(self, problema, prof_max=1000, inc_prof=1): 
        # Vai resolver um problema a cada profundidade até atingir a prof máxima
        # TODO: Talvez seja melhor fazer este incremento enquanto n encontrar uma solução
        for profundidade in range(inc_prof, prof_max + 1, inc_prof):
            solucao = super().resolver(problema, profundidade)
            return solucao
    